src/main/java/io/jenkins/plugins/gitlabbranchsource/servers/GitLabServer.java

```diff
@@ -0,0 +1,289 @@
+     * @param serverUrl     The URL of this GitLab Server
+     * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     * @param credentialsId The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for
+     *                      auto-management of hooks.
+     * @since 1.0.5
+     */
+    @DataBoundConstructor
+    public GitLabServer(@CheckForNull String displayName, @NonNull String serverUrl, boolean manageHooks,
+                       @CheckForNull String credentialsId) {
+        this.manageHooks = manageHooks;
+        this.credentialsId = credentialsId;
+        this.serverUrl = defaultIfBlank(serverUrl, GITLAB_SERVER_URL);
+        this.name = StringUtils.isBlank(displayName)
+                ? SCMName.fromUrl(this.serverUrl, COMMON_PREFIX_HOSTNAMES)
+                : displayName;
+    }
+
+
+    /**
+     * Looks up the {@link StandardCredentials} to use for auto-management of hooks.
+     *
+     * @return the credentials or {@code null}.
+     */
+    @CheckForNull
+    public StandardCredentials credentials() {
+        return StringUtils.isBlank(credentialsId) ? null : CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StandardCredentials.class,
+                        Jenkins.getActiveInstance(),
+                        ACL.SYSTEM,
+                        fromUri(serverUrl).build()
+                ),
+                CredentialsMatchers.allOf(
+                        AuthenticationTokens.matcher(GitLabAuth.class),
+                        withId(credentialsId)
+                )
+        );
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+
+    /**
+     * Our descriptor.
+     */
+    @Extension
+    public static class DescriptorImpl extends Descriptor<GitLabServer> {
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return "GitLab Server";
+        }
+
+
+        public String getAdvanceConfigMessage() {
+            return Messages.GitLabServer_advancedSectionForFuture();
+        }
+
+        /**
+         * Checks that the supplied URL is valid.
+         *
+         * @param value the URL to check.
+         * @return the validation results.
+         */
+        public static FormValidation doCheckServerUrl(@QueryParameter String value) {
+            // TODO: Add support for GitLab Ultimate (self hosted) and Gold (saas)
+            Jenkins.getActiveInstance().checkPermission(Jenkins.ADMINISTER);
+            try {
+                new URL(value);
+            } catch (MalformedURLException e) {
+                return FormValidation.error("Malformed GitLab url (%s)", e.getMessage());
+            }
+
+            if(GITLAB_SERVER_URL.equals(value)) {
```

Comment:
Nit: space between 'if' and '('

Reply:
Also, I'm not sure why this check is here. Does the URL always have to be gitlab.com? If so, we could hard code it instead of asking the user to enter it

Reply:
This check will extended to check is it is a url from a self-hosted GitLab server or GitLab gold. And I have set it to default value "https://gitlab.com" because it will be used commonly. We can change it in future based on user feedback.